### PR TITLE
Update Intro to Storybook for React in zh-TW

### PR DESCRIPTION
**Pull request by ymcheung**
- [x] Get Started
- [x] Simple Component
- [x] Composite Component
- [x] Data
- [x] Screens
- [x] Deploy
- [x] Testing
- [x] Addons
- [x] Conclusion
- [x] Contribute
- [x] Rebase

### CODE REVIEW

**Comment by jonniebigodes** [comment]
@ymcheung first and foremost. Thank you for the time and effort you put into this pull request and taking care of some language specific items in the configuration. If you don't mind and factor in the changes introduced by #456 we would really appreciate it. Let me know once you factored in and got the translation completed so that we can check in our community for a proofreader. Really stoked to see this version updated.

Stay safe

**Comment by ymcheung** [comment]
@jonniebigodes Hi, the translation into zh-TW for Intro to Storybook has completed, and gets rebased. There should be pull requests for corrections & updates later.

File: content/intro-to-storybook/react/zh-TW/composite-component.md [hunk 1]
```diff
@@ -164,87 +180,93 @@ function TaskList({ loading, tasks, onPinTask, onArchiveTask }) {
...
 
 it('renders pinned tasks at the start of the list', () => {
   const div = document.createElement('div');
-  const events = { onPinTask: jest.fn(), onArchiveTask: jest.fn() };
-  ReactDOM.render(<TaskList tasks={withPinnedTasks} {...events} />, div);
+  //👇 Story 測試要用到的 args
+  ReactDOM.render(<WithPinnedTasks {...WithPinnedTasks.args} />, div);
 
-  // 我們期望首先渲染標題為“任務6（固定）”的任務，而不是最後
+  // 期望標題是 "Task 6 (pinned)" 的任務先渲染出來，而不是在尾端
```
Comment (jonniebigodes) [inline content/intro-to-storybook/react/zh-TW/composite-component.md:256]: @ymcheung leaving this here but it applies to some other comments in the code snippets in the translation. If you don't mind leaving the snippets in their original form (English) we would appreciate it. I know that translating them would bring the "translation closer to home" (pardon the bad pun), but its something that has been applied to all translations currently available.

File: content/intro-to-storybook/react/zh-TW/conclusion.md [hunk 2]
```diff
@@ -1,31 +1,34 @@
...
+- [**視覺測試手冊**](https://storybook.js.org/tutorials/visual-testing-handbook)深入探討在 Storybook 以視覺測試元件。免費的 31 頁電子書。
 
-## 誰製作了 LearnStorybook.com?
+- 在 [**Discord 聊 Storybook**](https://discord.gg/UUt2PJb) 與 Storybook 社群進行互動。只要你幫助其他 Storybook 使用者，Storybook 使用者也會幫你。
 
-文字,程式碼和製作都是由[Chromatic](https://www.chromatic.com/). 該教程的靈感來自 Chromatic 的流行[GraphQL + React 教程系列](https://www.chromatic.com/blog/graphql-react-tutorial-part-1-6).
+- [**Storybook 部落格**](https://medium.com/storybookjs)展示能夠讓 UI 開發工作流程一致的最新版本與功能。
 
-想要更多這樣的教程和文章嗎? 註冊 Chromatic 郵件列表.
+## LearnStorybook.com 是誰做的？
```
Comment (jonniebigodes) [inline content/intro-to-storybook/react/zh-TW/conclusion.md:28]: @ymcheung if you could adjust the translated text to the equivalent of "Who made the Intro to Storybook tutorial?" we'd appreciate it!

File: content/intro-to-storybook/react/zh-TW/simple-component.md [hunk 3]
```diff
@@ -1,35 +1,32 @@
...
-
-我們將從基本實現開始,簡單傳入我們需要的`屬性-props` 以及 您可以對任務執行的兩個`on`操作 (在列表之間移動它) :
+一開始，先做好`工作事項`的基礎設定，只需帶入知道會用到的屬性，還有兩個在工作事項可以執行的動作（在清單之間移動）：
 
 ```javascript
+// src/components/Task.js
+
 import React from 'react';
 
 export default function Task({ task: { id, title, state }, onArchiveTask, onPinTask }) {
```
Comment (jonniebigodes) [inline content/intro-to-storybook/react/zh-TW/simple-component.md:32]: @ymcheung one thing regarding this chapter. I saw that you've incorporated the changes regarding the way snippets are displayed in other chapters but on this one. If you wouldn't mind adjusting the snippets here as you did in the other chapters we'd appreciate it.

For instance, the component's snippet would be represented as:

"```js:title=src/components/Task.js" (ignore the `"` they were introduced to render the desired format)
// the component's code snippet

**Review by jonniebigodes — CHANGES_REQUESTED** [review]
@ymcheung this is shaping up real nice! Before I ask around in the community for a proofreader to help us with this pull request I would like for you to look at a couple of items that require a bit of attention from your end! Let me know once you've taken care of them and we'll go from there.

Sounds good?

**Comment by ymcheung** [comment]
> Sounds good?

All done! :)

File: content/intro-to-storybook/react/zh-TW/simple-component.md [hunk 4]
```diff
@@ -41,95 +36,127 @@ export default function Task({ task: { id, title, state }, onArchiveTask, onPinT
...
+為了要讓 Storybook 知道正在撰寫文件的元件，先做好 `default` export，包含：
 
-要開始 Storybook,我們先運`行 註冊元件的`storiesOf()`函式. 我們為元件新增 _顯示名稱 - Storybook 應用程式側欄上顯示的名稱_.
+- `component` -- 元件它本人；
+- `title` -- 在 Storybook app 側邊欄找到元件的名字；
+- `excludeStories` -- story 檔案裡的 export，但不應被 Storybook render 為 story；
+- `argTypes` -- 設定每個 story 的[參數](https://storybook.js.org/docs/react/api/argtypes)行為。
 
-`action()`允許我們建立一個回撥, 當在 Storybook UI 的面板中 單擊這個 **action** 時 回撥觸發. 因此,當我們構建一個 pin 按鈕 時,我們將能夠在 測試 UI 中 確定按鈕單擊 是否成功.
+若要定義出各式各樣的 story，就為每個要測試的狀態 export 函式，以此產生 story。Story 就是根據設定好的狀態，回傳 render 出元素的函式（也就是有一組 prop 的元件），這跟 [Stateless Functional Component](https://reactjs.org/docs/components-and-props.html) 一樣。
```
Comment (jonniebigodes) [inline content/intro-to-storybook/react/zh-TW/simple-component.md:96]: @ymcheung if you wouldn't mind removing the "stateless" from "Stateless functional component". As technically  it's deprecated.

File: content/intro-to-storybook/react/zh-TW/simple-component.md [hunk 5]
```diff
@@ -41,95 +36,127 @@ export default function Task({ task: { id, title, state }, onArchiveTask, onPinT
...
-關於捆綁`actions`的另一個好處就是,你可以`export-暴露`它們,用於重用該元件的元件,我們稍後會看到.
+<div class="aside">
+💡 <code>Template.bind({})</code> 是 <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/bind">標準的 JavaScript</a> 技巧，用來複製函式。此技巧是用來讓每個 export 出來的 story 可以設定各自的 property，但使用完全相同的方法。
+</div>
 
-為了定義我們的故事,我們用`add()`,一次一個為我們的每個測試狀態生成一個故事. `add`第二個引數是一個函式,它返回一個給定狀態的渲染元素 (即帶有一組`props`的元件類) - 就像一個 React[無狀態功能元件](https://reactjs.org/docs/components-and-props.html).
+參數，或簡稱為 [`args`](https://storybook.js.org/docs/react/writing-stories/args)，可以透過 Controls 外掛即時更新元件，而且不用重啟 Storybook。一旦 [`args`](https://storybook.js.org/docs/react/writing-stories/args) 值改變了，元件也跟著變。
 
-在建立故事時,我們使用基本任務 (`task`) 構建元件期望的 任務的形狀. 這通常是 根據真實資料的模型建模的. 再次,正如我們所看到的,`export`這種形狀將使我們能夠在以後的故事中重複使用它.
+在新增 story 的時候，會用到基本的 `task` 參數，來蓋出元件應該要有的工作事項外觀。This is typically modelled from what the true data looks like. 這裡還是一樣，以這樣的結構來輸出，讓我們可以在之後的做的 story 重複使用，接下來還會有。
```
Comment (jonniebigodes) [inline content/intro-to-storybook/react/zh-TW/simple-component.md:106]: @ymcheung did you forget to translate this sentence? "This is typically modeled from what the true data looks like." ????

File: content/intro-to-storybook/react/zh-TW/simple-component.md [hunk 6]
```diff
@@ -41,95 +36,127 @@ export default function Task({ task: { id, title, state }, onArchiveTask, onPinT
...
-`action()`允許我們建立一個回撥, 當在 Storybook UI 的面板中 單擊這個 **action** 時 回撥觸發. 因此,當我們構建一個 pin 按鈕 時,我們將能夠在 測試 UI 中 確定按鈕單擊 是否成功.
+若要定義出各式各樣的 story，就為每個要測試的狀態 export 函式，以此產生 story。Story 就是根據設定好的狀態，回傳 render 出元素的函式（也就是有一組 prop 的元件），這跟 [Stateless Functional Component](https://reactjs.org/docs/components-and-props.html) 一樣。
 
-由於我們需要將 相同的一組操作 傳遞給 元件的所有排列,因此將它們捆綁到`actions`變數 並 使用 React`{...actions}`的`porps`擴充套件以立即傳遞它們. `<Task {...actions}>`相當於`<Task onPinTask={actions.onPinTask} onArchiveTask={actions.onArchiveTask}>`.
+既然已經排列出元件，賦值給稱為 `Template` 的變數會很方便。採用這樣的模式來做 story 可以減少撰寫和維護程式碼的份量。
 
-關於捆綁`actions`的另一個好處就是,你可以`export-暴露`它們,用於重用該元件的元件,我們稍後會看到.
+<div class="aside">
+💡 <code>Template.bind({})</code> 是 <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/bind">標準的 JavaScript</a> 技巧，用來複製函式。此技巧是用來讓每個 export 出來的 story 可以設定各自的 property，但使用完全相同的方法。
+</div>
```
Comment (jonniebigodes) [inline content/intro-to-storybook/react/zh-TW/simple-component.md:102]: @ymcheung Looking at this aside, you can translate it at will. Minus the "story", which in this case relates to Storybook's stories, the rest is ok.

**Review by jonniebigodes — CHANGES_REQUESTED** [review]
@ymcheung I left a couple of items that you might need to adjust on your end. In the meantime, I'm going to reach out and see if anyone is willing to help you and us get it proofread and merged asap.

As I've mentioned to another contributor. If no one comes forward I'll gladly merge this.

Stay safe

**Comment by ymcheung** [comment]
> @ymcheung I left a couple of items that you might need to adjust on your end. In the meantime, I'm going to reach out and see if anyone is willing to help you and us get it proofread and merged asap.
> 
> As I've mentioned to another contributor. If no one comes forward I'll gladly merge this.

@jonniebigodes I've just got them done!

**Comment by ymcheung** [comment]
@jonniebigodes I rebased and updated the content for zh-TW.

**Comment by ymcheung** [comment]
@jonniebigodes It's been a while since this PR asked for community proofreading. I didn't receive any comments about the translation. What about merging this PR so that a wider range of people can read/suggest if there's no technical issue?

**Comment by jonniebigodes** [comment]
@ymcheung factor in the changes in #504 and I'll gladly merge it. It's been too long and no one came forward.

**Comment by ymcheung** [comment]
> @ymcheung factor in the changes in #504 and I'll gladly merge it. It's been too long and no one came forward.

Done :)

**Comment by jonniebigodes** [comment]
@ymcheung thanks for taking care of the last item, and I'm with you. No one came forward to proofread this and this has waited long enough. I'm going to merge this and if anyone should be interested in an update, they can follow up with a pull request.

One final thing, if you could jump into our [Discord Server](https://discord.gg/UUt2PJb) and message me (same username), as there's one last item that I want to talk to you about. Sounds good?

And it goes without saying, thank you very much for the time and effort you've put into this pull request and sorry for the wait.

Stay safe

**Pull request merged**